### DISCUSSION TITLE
Support for new TV series scanner

### DISCUSSION THREAD

**Discussion by flammable** [Taxel/PlexTraktSync#144] [Q&A] [answered]
Hey guys -

I know this is brand new and still in beta, so my hunch is that it's not supported yet - if not, I'll hold off from switching my TV show libraries.

https://forums.plex.tv/t/beta-new-plex-tv-series-scanner/696242

I didn't see any discussions or issues on this. Sorry if it's already been covered!

Thanks for your work on this script!

**Comment by glensc** [comment]
@flammable we rely on PlexAPI library:
- https://pypi.org/project/PlexAPI/

so, check if they have support for it, and update it here.

**Reply by flammable** [reply]
Thanks! I'm not sure why I checked the Trakt side of things instead of Plex. 🤦🏻‍♂️

It looks like support was added in [PlexAPI 4.4.1](https://github.com/pkkid/python-plexapi/releases/tag/4.4.1), which is the version in [requirements.txt](https://github.com/Taxel/PlexTraktSync/blob/15acdea06bfa1d76dab10c09fd149e05e17abf91/requirements.txt) - so we should be good to go, right?

I converted my movie library to the new scanner way too early, so I couldn't fully use this script for a few months. I didn't want to make the same mistake with the TV scanner.

**Reply by glensc** [reply]
edited your post, because 'never link to branches'

**Reply by wandabastyle** [reply]
I added the code changes from movie section to the show section and it seems to work. Only problem I have is that episodes that I watch a second time are not represented in trakt history.

**Reply by flammable** [reply]
@wandabastyle It'd be really awesome if you're willing to submit a pull request for that! It sounds like #152 was a step towards recording multiple views for Plex media.

**Reply by wandabastyle** [reply]
In `def process_show_section` below `guid = show.guid` I added:

```
if guid.startswith('plex://show/'):
    if len(show.guids) > 0:
    logging.debug("trying first alternative guid: " + str(show.guids[0].id))
        guid = show.guids[0].id
    x = provider = None
```
And after the next `continue`:

```
elif 'imdb' in guid:
    x = guid.split('//')[1]
    x = x.split('?')[0]
    provider = 'imdb'
```

It's the same change which was made to the movie section previously, but it works.
Dunno if this is the way to go as I just copied it. Also I don't know how this impacts on performance.

Sorry about the formatting, I have no idea how the github editor works.

**Reply by flammable** [reply]
Thanks for this! I hope it's incorporated it into a release!

**Reply by glensc** [reply]
@flammable yes:

1. 0.5.x: https://github.com/Taxel/PlexTraktSync/pull/172 (unfinished)
2. 0.6.x & 0.7.x: https://github.com/Taxel/PlexTraktSync/pull/172#issuecomment-817150928

**Comment by glensc** [comment, marked as answer]
This is solved:
- https://github.com/Taxel/PlexTraktSync/discussions/144#discussioncomment-611277

unfortunately, discussions can't be "closed", they will stay "open" forever.

**Reply by glensc** [reply]
discussions can be closed now. closed it.

**Reply by flammable** [reply]
I think this might still be open, as I can reply.

**Reply by glensc** [reply]
to prevent replies it's called locking.
